pkg/controller/endpointslicemirroring/reconciler_test.go

```diff
@@ -837,6 +947,8 @@ func TestReconcile(t *testing.T) {
 				expectMetrics(t, *tc.expectedMetrics)
 			}
 
+			expectAnnotations(t, endpoints, tc.existingEndpointSlices)
```

Comment:
This is not actually what you want here. You're comparing the two test inputs (Endpoints and EndpointSlices that exist before controller syncs), not the actual output (fetched below).

Reply:
Actually, I think you could just expand `expectEndpointSlices` to always expect this annotation to be unset. This would be similar to how it always expects the service name label to be set: https://github.com/kubernetes/kubernetes/blob/576457ac2508d2e06782805f9b3cb27dbbcf7324/pkg/controller/endpointslicemirroring/reconciler_test.go#L988-L991.

Reply:
I see my mistake. Thank you for the pointers. I'll update my PR

Reply:
@robscott 
If i understand you correctly, all that should be needed at the line you specified inside `reconciler_test.go` is:
```
if epSlice.Annotations[corev1.LastAppliedConfigAnnotation] != "" {
        t.Errorf("Expected LastAppliedConfig Annotation to be unser, got %s", epSlice.Annotations[corev1.LastAppliedConfigAnnotation])
}
```

Reply:
Yep, exactly something like that. We'd also need some way for a test case to actually set that annotation on the original Endpoints and EndpointSlice resources as well. You've already got the EndpointSlice use case covered, but it will be a bit more difficult to do the same for the Endpoints in the test case since we just have a `subsets` object instead of the full resource. Maybe just add a new `endpointsAnnotations` struct to the testCase struct?

Reply:
Yes, a map[string]string should work for that, I believe?

Reply:
Yep, that sounds ideal.